src/app/data-planes/data/index.ts

```diff
@@ -84,24 +170,13 @@ export function getTags(dataplaneOverview: DataplaneOverview): LabelValue[] {
   })
 }
 
-export function getIsCertExpired(dataplaneOverview: DataplaneOverview): boolean {
-  const mTLS = dataplaneOverview.dataplaneInsight?.mTLS
-
-  if (mTLS === undefined) {
-    return false
-  }
-
-  return Date.now() > new Date(mTLS.certificateExpirationTime).getTime()
+function getIsCertExpired({ mTLS }: DataplaneInsight): boolean {
+  return mTLS ? Date.now() > new Date(mTLS.certificateExpirationTime).getTime() : false
 }
 
-export function getWarnings(dataplaneOverview: DataplaneOverview, canUseZones: boolean): DataplaneWarning[] {
-  const subscriptions = dataplaneOverview.dataplaneInsight?.subscriptions ?? []
-  if (subscriptions.length === 0) {
-    return []
-  }
-
-  const lastSubscription = subscriptions[subscriptions.length - 1]
-  if (!('version' in lastSubscription) || !lastSubscription.version) {
+function getWarnings({ subscriptions }: DataplaneInsight, tags: LabelValue[], canUseZones: boolean): DataplaneWarning[] {
+  const lastSubscription = subscriptions.at(-1)
```

Comment:
Similar to above this should use `connectedSubscription`

Reply:
I don’t think so. Why should we use `connectedSubscription` here? This information is also useful if the Dataplane is disconnected.

Reply:
As far as I understand it, all subscriptions but the latest one are irrelevant when it comes to presenting current information about a Dataplane. None of its data might still be in effect so showing warnings based on that would be incorrect.

Reply:
Yes, that’s how I understand it. Overall I think it’s more useful to show the same kinds of warnings for a disconnected Dataplane that you would for a connected one.

Reply:
Ok gotcha, thanks, similar to the over thread on this PR I think there'll be a bit of work off the back of this. Feel free to resolve this one.

Reply:
Other related thread for reference: https://github.com/kumahq/kuma-gui/pull/1871#discussion_r1422451135